src/rpc/blockchain.cpp

```diff
@@ -1515,8 +1514,18 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
         }
     }
 
-    if (blockcount < 1 || blockcount >= pindex->nHeight) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block count: should be between 1 and the block's height");
+    if (pindex->nHeight < 1) {
+        blockcount = 0;
+    } else {
+        blockcount = std::min(blockcount, pindex->nHeight - 1);
+    }
+
+    if (!request.params[0].isNull()) {
+        blockcount = request.params[0].get_int();
+
+        if (blockcount < 0 || (blockcount > 0 && blockcount >= pindex->nHeight)) {
```

Comment:
@promag @jnewbery I think that this condition is necessary if pindex points to `genesis block`.